pkg/codegen/internal/test/testdata/provider-config-schema/python/pulumi_configstation/config/vars.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# *** WARNING: this file was generated by test. ***
+# *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import warnings
+import pulumi
+import pulumi.runtime
+from typing import Any, Mapping, Optional, Sequence, Union, overload
+from .. import _utilities
+from . import outputs
+from .. import outputs as _root_outputs
+
+import types
+
+__config__ = pulumi.Config('configstation')
+
+
+class _ExportableConfig(types.ModuleType):
+    @property
+    def favorite_plants(self) -> Optional[Sequence[str]]:
+        return __config__.get('favoritePlants')
+
+    @property
+    def favorite_sandwich(self) -> Optional[pulumi.InputType['outputs.Sandwich']]:
+        """
+        omg my favorite sandwich
+        """
+        return __config__.get('favoriteSandwich')
```

Comment:
Two issues:

1. If we want to return the object, it should be typed as `Optional['outputs.Sandwich']` (remove the `pulumi.InputType`)
2. It's going to need to use `get_object`, but that alone isn't enough. `get_object` uses `json.loads(v)` to deserialize the JSON string to an object, but this'll just return a `dict`, so we'll need to convert the dict to the output type.
    - For inputs, we have a helper ([`pulumi.runtime.to_json`](https://github.com/pulumi/pulumi/blob/a952140b34c667d005116ee53be08077b5863948/sdk/python/lib/pulumi/runtime/_json.py#L40)) that helps serialize an input type to JSON, which is used by the `Provider` resource class (e.g. [in AWS](https://github.com/pulumi/pulumi-aws/blob/1b4fa4fecdc5b89c0141f07dcc9f65a4a1465d85/sdk/python/pulumi_aws/provider.py#L456)).
    - Perhaps we need a similar `pulumi.runtime.from_json` that can be used here, if we want to return these objects. It'd have to work similar to how we unmarshal RPC values into outputs, to instantiate the right output types based on the type information.
        - We might need `pulumi.runtime.from_json` anyway for https://github.com/pulumi/pulumi/issues/6259, if we want to expose the non-string outputs on the `Provider` resource, if we're not doing the serialization/deserialization at the Go -> gRPC boundary.

Reply:
Interesting... yeah I was suspicious of the types that it was generating, especially the `pulumi.InputType`. I wonder if this is why the [existing config variables](https://github.com/pulumi/pulumi-aws/blob/master/sdk/python/pulumi_aws/config/vars.py) don't specify types at all.